src/main/java/org/jabref/gui/externalfiles/FindFullTextAction.java

```diff
@@ -96,18 +97,24 @@ public void update() {
                 DownloadExternalFile fileDownload = new DownloadExternalFile(basePanel.frame(),
                         basePanel.getBibDatabaseContext(), entry);
                 try {
-                    fileDownload.download(result.get(), "application/pdf", file -> {
-                        DefaultTaskExecutor.runInJavaFXThread(() -> {
-                            Optional<FieldChange> fieldChange = entry.addFile(file);
-                            if (fieldChange.isPresent()) {
-                                UndoableFieldChange edit = new UndoableFieldChange(entry, FieldName.FILE,
-                                        entry.getField(FieldName.FILE).orElse(null), fieldChange.get().getNewValue());
-                                basePanel.getUndoManager().addEdit(edit);
-                                basePanel.markBaseChanged();
-                            }
-                        });
+ 		    String str = dir.toString() + entry.getField(FieldName.FILE);
```

Comment:
This won't work because dir is an optional. You would need get() first.
Also, File class should be avoided. Better use the path class and the other classes from the nio package:
https://docs.oracle.com/javase/tutorial/essential/io/fileio.html

And your code does not really solve the problem stated in the issue.
The idea is that JabRef previously searched for the file in all directories and linked it, if found.
If not found, it would download the file.

So the idea here would be to first link local files. This is already implemented in the AutoSetFileLinksUtil. 
This could be reused or the method called before.